pkg/thanos/operator.go

```diff
@@ -689,11 +691,12 @@ func (o *Operator) UpdateStatus(ctx context.Context, key string) error {
 	availableCondition := stsReporter.Update(tr)
 	reconciledCondition := o.reconciliations.GetCondition(key, tr.Generation)
 	tr.Status.Conditions = operator.UpdateConditions(tr.Status.Conditions, availableCondition, reconciledCondition)
-
 	tr.Status.Paused = tr.Spec.Paused
 
-	if _, err = o.mclient.MonitoringV1().ThanosRulers(tr.Namespace).UpdateStatus(ctx, tr, metav1.UpdateOptions{}); err != nil {
-		return errors.Wrap(err, "failed to update status subresource")
+	trac := ApplyConfigurationFromThanosRuler(tr)
```

Comment:
(nit) the `trac` variable is a bit superfluous, the call to `ApplyConfigurationFromThanosRuler()` can be inlined.

Reply:
yep, I just don't want to do so long the next func